pkg/controllers/managementagent/monitoring/operatorHandler.go

```diff
@@ -40,32 +40,37 @@ func (h *operatorHandler) syncCluster(key string, obj *mgmtv3.Cluster) (runtime.
 		return obj, nil
 	}
 
-	var newCluster *mgmtv3.Cluster
-	var err error
+	cpy := obj.DeepCopy()
+	var err, updateErr error
 	//should deploy
 	if obj.Spec.EnableClusterAlerting || obj.Spec.EnableClusterMonitoring {
-		newObj, err := v32.ClusterConditionPrometheusOperatorDeployed.Do(obj, func() (runtime.Object, error) {
-			cpy := obj.DeepCopy()
+		newObj, err := v32.ClusterConditionPrometheusOperatorDeployed.Do(cpy, func() (runtime.Object, error) {
 			return cpy, deploySystemMonitor(cpy, h.app, h.catalogManager, h.clusterName)
 		})
 		if err != nil {
 			logrus.Warnf("deploy prometheus operator error, %v", err)
 		}
-		newCluster = newObj.(*mgmtv3.Cluster)
+		cpy = newObj.(*mgmtv3.Cluster)
 	} else { // should withdraw
-		newCluster = obj.DeepCopy()
-		if err = withdrawSystemMonitor(newCluster, h.app); err != nil {
+		if err = withdrawSystemMonitor(cpy, h.app); err != nil {
 			logrus.Warnf("withdraw prometheus operator error, %v", err)
 		}
 	}
 
-	if newCluster != nil && !reflect.DeepEqual(newCluster, obj) {
-		if newCluster, err = h.clusters.Update(newCluster); err != nil {
-			return nil, err
-		}
-		return newCluster, nil
+	if !reflect.DeepEqual(cpy, obj) {
+		cpy, updateErr = h.clusters.Update(cpy)
```

Comment:
Use `RetryOnConflict` on here https://github.com/rancher/rancher/blob/master/pkg/controllers/management/auth/crtb_handler.go#L225
Then if it still fails use `multierror.Append` to return both errors.